misfitapp/models.py

```diff
@@ -2,17 +2,72 @@
 from django.db import models, IntegrityError
 from django.utils.encoding import python_2_unicode_compatible
 from math import pow
+from misfit.notification import MisfitMessage
 import datetime
 
-from .extras import (
-    cc_to_underscore_keys,
-    chunkify_dates,
-    dedupe_by_field)
-
+DAYS_IN_CHUNK = 30
 MAX_KEY_LEN = 24
+MISFIT_HISTORIC_TIMEDELTA = getattr(settings, 'MISFIT_HISTORIC_TIMEDELTA',
+                                    datetime.timedelta(days=90))
+HISTORIC_START_DATE = datetime.date.today() - MISFIT_HISTORIC_TIMEDELTA
 UserModel = getattr(settings, 'AUTH_USER_MODEL', 'auth.User')
 
 
+def chunkify_dates(start, end, days_in_chunk=DAYS_IN_CHUNK):
+    """
+    Return a list of tuples that chunks the date range into ranges
+    of length days_in_chunk, inclusive of the end date. So the end
+    date of one chunk is equal to the start date of the chunk after.
+    """
+    chunks = []
+    s = start
+    e = start + datetime.timedelta(days=days_in_chunk)
+    while e - datetime.timedelta(days=days_in_chunk) < end:
+        e = min(e, end)
+        chunks.append((s, e))
+        s = e
+        e = s + datetime.timedelta(days=days_in_chunk)
+    return chunks
+
+
+def dedupe_by_field(l, field):
+    """
+    Returns a new list with duplicate objects removed. Objects are equal
+    iff the have the same value for 'field'.
+    """
+    d = dict((getattr(obj, field), obj) for obj in l)
+    return list(d.values())
+
+
+class MisfitModel(models.Model):
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def process_message(cls, message, misfit, uid):
```

Comment:
@brad Nice refactor here. I like having the models inherit from a base class and all use this logic.